### ISSUE TITLE
Opening hdf files in VisGUI on clean windows/conda install fails with most recent windows conda package

### ISSUE THREAD

**Issue by David-Baddeley** [python-microscopy/python-microscopy#60]
**[Original report](https://bitbucket.org/david_baddeley/python-microscopy/issue/60) by Andrew Barentine (Bitbucket: [AndrewBarentine](https://bitbucket.org/AndrewBarentine), ).**

----------------------------------------

I can open the same hdf file on OSX without problem.

```
#!python

Traceback (most recent call last):

  File "C:\ProgramData\Anaconda2\lib\site-packages\wx-3.0-msw\wx\_misc.py", line 1367, in Notify

    self.notify()

  File "C:\ProgramData\Anaconda2\lib\site-packages\wx-3.0-msw\wx\_core.py", line 16870, in Notify

    self.result = self.callable(*self.args, **self.kwargs)

  File "C:\ProgramData\Anaconda2\lib\site-packages\PYME\LMVis\visCore.py", line 665, in OpenFile

    self.pipeline.OpenFile(filename, **args)

  File "C:\ProgramData\Anaconda2\lib\site-packages\PYME\LMVis\pipeline.py", line 668, in OpenFile

    self.ev_mappings, self.eventCharts = _processEvents(mapped_ds, self.events, self.mdh)

  File "C:\ProgramData\Anaconda2\lib\site-packages\PYME\LMVis\pipeline.py", line 247, in _processEvents

    'StackSettings.StepSize'], mdh['StackSettings.StepSize'])

ValueError: arange: cannot compute length
```

**Comment by David-Baddeley** [comment]
**Original comment by David Baddeley (Bitbucket: [david_baddeley](https://bitbucket.org/david_baddeley), ).**

----------------------------------------

That error message is really odd (as I don't see the call to arange anywhere)

I've just rebuilt the windows packages as they were fairly out of date. Could you try again and see if that helps things?

**Comment by David-Baddeley** [comment]
**Original comment by Andrew Barentine (Bitbucket: [AndrewBarentine](https://bitbucket.org/AndrewBarentine), ).**

----------------------------------------

Thanks David. I'll try on Wednesday when I meet with Joerg (we were trying on his laptop).

**Comment by David-Baddeley** [comment]
**Original comment by David Baddeley (Bitbucket: [david_baddeley](https://bitbucket.org/david_baddeley), ).**

----------------------------------------

It might be nice to see if we can try this out beforehand - could you send me a copy of the file?

**Comment by David-Baddeley** [comment]
**Original comment by David Baddeley (Bitbucket: [david_baddeley](https://bitbucket.org/david_baddeley), ).**

----------------------------------------

OK, this boils down to an error when attempting to spoof event data from the `StackSettings` metadata, which is in turn due to an error in the StackSettings metadata (a step size of 0). This is in some code you added early in 2017.

My preference would be to remove this completely from the pipeline and instead move the logic to the spooler so that we get a legit events dataset. We certainly don't want this code running when we load .h5 as any mapping should have been done at the .h5r stage. Even at the .h5r stage, this probably shouldn't run automatically, as it indicates that there is something wrong with the input (no events). If necessary we could shift the logic to an optional Extras module that allowed you to do this after loading the data.

**Comment by David-Baddeley** [comment]
**Original comment by David Baddeley (Bitbucket: [david_baddeley](https://bitbucket.org/david_baddeley), ).**

----------------------------------------

I'm not sure why the bug only effects windows, however. The OSX version crashes out a couple of lines later (but gets caught and handled).

**Comment by David-Baddeley** [comment]
**Original comment by David Baddeley (Bitbucket: [david_baddeley](https://bitbucket.org/david_baddeley), ).**

----------------------------------------

I've put a try-catch around the whole lot as a temporary fix for loading under windows (commit 835591d)

**Comment by David-Baddeley** [comment]
**Original comment by David Baddeley (Bitbucket: [david_baddeley](https://bitbucket.org/david_baddeley), ).**

----------------------------------------

fixed in 835591d
